WordPress/src/main/java/org/wordpress/android/ui/mlp/ModalLayoutPickerListDiffCallback.kt

```diff
@@ -9,10 +15,24 @@ class ModalLayoutPickerListDiffCallback(
     private val oldList: List<ModalLayoutPickerListItem>,
     private val newList: List<ModalLayoutPickerListItem>
 ) : Callback() {
+    object Payload
+
     override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
         val newItem = newList[newItemPosition]
         val oldItem = oldList[oldItemPosition]
-        return oldItem == newItem
+        if (newItem.type == oldItem.type) {
+            return when (newItem.type) {
+                TITLE -> (newItem as? Title)?.visible == (oldItem as? Title)?.visible
+                SUBTITLE -> true
+                CATEGORIES -> true // for now
+                LAYOUTS -> {
+                    val newLayoutCategory = (newItem as? LayoutCategory)
+                    val oldLayoutCategory = (oldItem as? LayoutCategory)
+                    return newLayoutCategory?.description == oldLayoutCategory?.description &&
```

Comment:
Similar to titles - if two categories have different descriptions or sizes they can't be recycled?

Reply:
As above fixed with baee77e672b474f115007ee6ae593172d4065660